### PR TITLE
Adds functionality to strength and stimmed

### PR DESCRIPTION
**Pull request by trollbreeder**
## About The Pull Request

This adds functionality to strength and stimmed, so they're not just craftgenes.
Strength allows you to carry mobs and lockers without being slowed down. Very useful for would-be paramedics, as long as you stop the bleeding. Nyoom.
Stimmed is what i hope will become used often, maybe even a must-have like space adap. It removes toxic reagents slowly but consistently from the user. Dap on the chemist as you shrug off their chloral hydrate cyanide nitric acid fluorosulfuric acid lexorin deathmix! Drink sulfonal as a party trick! Etc etc..

## Why It's Good For The Game

Fun new mutations to play with, no more throwing away these craftgenes after being used to make a better mutation. Now they have actual purpose to be injected into you, which is good.

## Changelog
:cl:
add: The strength mutation now allows dragging heavy objects with ease.
add: The stimmed mutation now slowly purges toxic reagents.
balance: Stimmed and strength have been given instability
/:cl:

### CODE REVIEW

**Comment by trollbreeder** [comment]
hold on a second

**Comment by Pumpkinoe** [comment]
woah!!! this is cool and good
does strength stop all slowdown from carrying? dragging?

**Comment by trollbreeder** [comment]
It stops slowdown from pulling lockers and mobs, since those use drag_slowdown. Carrying is not handled under this, i think.

**Comment by Pumpkinoe** [comment]
maybe this will pave the way to making gigantism have functionality too
strength + giantism when?

**Comment by Pugie** [comment]
There's a reason these slowdowns exist - because being dragged away by someone murderboning on meth without other people being able to save you is lame.

**Comment by TiviPlus** [comment]
Dragfighting is back in town

File: code/datums/mutations/body.dm [hunk 1]
```diff
@@ -224,17 +224,35 @@
...
+
+/datum/mutation/human/strong/on_acquiring(mob/living/carbon/human/owner)
+	if(..())
+		return
+	owner.slowed_by_drag = FALSE // Don't get slowed by dragging lockers or mobs
+
+/datum/mutation/human/strong/on_losing(mob/living/carbon/human/owner)
+	if(..())
+		return
+	owner.slowed_by_drag = initial(owner.slowed_by_drag)
```
Comment (PKPenguin321) [inline code/datums/mutations/body.dm]: slowed_by_drag should be made a trait for futureproofing

**Comment by PKPenguin321** [comment]
imo Strength will not be worth the 25 instability 99% of the time, seeing as if you have it you're half of the way to having hulk and hulk is just straight up better use of your genetic capacity if you want to murderbone. it's nothing to really worry about

**Comment by necromanceranne** [comment]
PKP,  only humans can be hulks.

**Comment by trollbreeder** [comment]
So, 20 instability instead?

File: code/datums/mutations/body.dm (hunk 1 above)
Comment (trollbreeder) [reply code/datums/mutations/body.dm]: i'll look into it

**Comment by MimicFaux** [comment]
A decent instability would be worthwhile to balance against the reason slowdowns were introduced in the firstplace.

**Comment by trollbreeder** [comment]
Slowed_by_drag refactored into a trait, and instability is now 20, from 25.

**Comment by ATH1909** [comment]
Can you make strength also make you ignore the slowdown from fireman carrying someone?

**Comment by trollbreeder** [comment]
> Can you make strength also make you ignore the slowdown from fireman carrying someone?

I tried coding that, but i tested it and it doesn't seem like it works. The slowdown from fireman carrying is negligible in my testing.

File: code/datums/mutations/body.dm [hunk 2]
```diff
@@ -224,17 +224,35 @@
...
 	quality = POSITIVE
-	text_gain_indication = "<span class='notice'>You feel strong.</span>"
+	text_gain_indication = "<span class='notice'>You feel strong!</span>"
 	difficulty = 16
+	instability = 20
+
+/datum/mutation/human/strong/on_acquiring(mob/living/carbon/human/owner)
+	if(..())
+		return
+	ADD_TRAIT(owner, TRAIT_NOT_SLOWEDBYDRAG, "genetics") // Don't get slowed by dragging lockers or mobs
```
Comment (Rohesie) [inline code/datums/mutations/body.dm]:
```suggestion
	owner.add_movespeed_mod_immunities(type, /datum/movespeed_modifier/bulky_drag)
	owner.add_movespeed_mod_immunities(type, /datum/movespeed_modifier/human_carry)
```

File: code/datums/mutations/body.dm [hunk 3]
```diff
@@ -224,17 +224,35 @@
...
+
+/datum/mutation/human/strong/on_losing(mob/living/carbon/human/owner)
+	if(..())
+		return
+	REMOVE_TRAIT(owner, TRAIT_NOT_SLOWEDBYDRAG, "genetics")
```
Comment (Rohesie) [inline code/datums/mutations/body.dm]: ```suggestion```suggestion
	owner.remove_movespeed_modifier(type, /datum/movespeed_modifier/bulky_drag)
	owner.remove_movespeed_modifier(type, /datum/movespeed_modifier/human_carry)
```
```

File: code/datums/mutations/body.dm [hunk 4]
```diff
@@ -224,17 +224,35 @@
...
-	desc = "The user's muscles slightly expand."
+	desc = "The user's muscles slightly expand, allowing them to move heavy objects easily."
 	quality = POSITIVE
-	text_gain_indication = "<span class='notice'>You feel strong.</span>"
+	text_gain_indication = "<span class='notice'>You feel strong!</span>"
 	difficulty = 16
+	instability = 20
+
+/datum/mutation/human/strong/on_acquiring(mob/living/carbon/human/owner)
+	if(..())
```
Comment (Rohesie) [inline code/datums/mutations/body.dm]:
```suggestion
	. = ..()
	if(.)
```

File: code/datums/mutations/body.dm (hunk 3 above)
Comment (trollbreeder) [reply code/datums/mutations/body.dm]: you scuffed this suggestion

File: code/datums/mutations/body.dm (hunk 2 above)
Comment (trollbreeder) [reply code/datums/mutations/body.dm]: ah, thanks, that's what i was looking for when i was trying to add carry slowdown immunity

File: code/datums/mutations/body.dm [hunk 5]
```diff
@@ -224,17 +224,36 @@
...
+
+/datum/mutation/human/strong/on_acquiring(mob/living/carbon/human/owner)
+	. = ..()
+	if(.)
+		return
+	ADD_TRAIT(owner, TRAIT_NOT_SLOWEDBYDRAG, "genetics") // Don't get slowed by dragging lockers or mobs
+
+/datum/mutation/human/strong/on_losing(mob/living/carbon/human/owner)
+	if(..())
+		return
```
Comment (Rohesie) [inline code/datums/mutations/body.dm]:
```suggestion
	. = ..()
	if(.)
		return
```

**Comment by trollbreeder** [comment]
add_movespeed_mod_immunities is an unknown proc

**Comment by Rohesie** [comment]
Update your code.

**Comment by trollbreeder** [comment]
I'm doing so

**Comment by trollbreeder** [comment]
@Cyberboss Help, the Appveyor gods are angry and they say they cannot download GitHub repository contents because it is forbidden

**Comment by Donpedrito** [comment]
@pumpkinoe Gigantism increases your odds of tackling successfully

**Comment by trollbreeder** [comment]
i have no fucking idea what i'm doing on the github side and i'm thinking of closing this and making a sequel PR

**Comment by trollbreeder** [comment]
Remaking this from an updated master. Keep an eye out for the new PR!

**Pull request closed without merging**